Neo4j.Driver/Neo4j.Driver/Neo4jException.cs

```diff
@@ -464,6 +464,110 @@ public InvalidBookmarkException(string message) : base(ErrorCode, message)
         }
     }
 
+    /// <summary>
+    /// The provided bookmark is invalid. To recover from this a new session needs to be created.
+    /// </summary>
+    public class InvalidBookmarkMixtureException : ClientException
+    {
+        private const string ErrorCode = "Neo.ClientError.Transaction.InvalidBookmarkMixture";
+
+        /// <summary>
+        /// Create a new <see cref="InvalidBookmarkMixtureException" /> with an error message.
+        /// </summary>
+        /// <param name="message">The error message.</param>
+        public InvalidBookmarkMixtureException(string message) : base(ErrorCode, message)
+        {
+        }
+
+        internal static bool IsInvalidBookmarkMixtureException(string code)
+        {
+            return string.Equals(code, ErrorCode);
+        }
+    }
+
+    /// <summary>
+    /// A generic argument error has occurred. To recover from this a new session needs to be created.</summary>
+    [DataContract]
+    public class ArgumentErrorException : ClientException
```

Comment:
Maybe this should just be called `ArgumentException` to be more in line with the others. The code `Neo.ClientError.*` gets hydrated into a `ClientException` after all (as opposed to `ClientErrorException`).

(same for the other exception classes you added)